ark/util/arrays.ts

```diff
@@ -3,6 +3,34 @@ import type { anyOrNever, conform } from "./generics.ts"
 import type { isDisjoint } from "./intersections.ts"
 import type { parseNonNegativeInteger } from "./numbers.ts"
 
+type DuplicateData<val = unknown> = { element: val; indices: number[] }
+
+/**
+ * Extracts duplicated elements and their indices from an array, returning them.
+ *
+ * @param arr The array to extract duplicate elements from.
```

Comment:
Nice! Thanks for adding as a util here. I guess this is where we could make the change to include the initial index?

Reply:
Yep, done in cc44db7 (and even fixed a bug in the process!)